jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -298,9 +385,7 @@ public boolean pushBlob(
       Consumer<Long> writtenByteCountListener)
       throws IOException, RegistryException {
 
-    if (sourceRepository != null
-        && authorization != null
-        && !authorization.canAccess(sourceRepository, "pull")) {
+    if (sourceRepository != null && canMountBlobs(authorization, sourceRepository)) {
```

Comment:
I think the condition is reversed. `... && !canMountBlobs()`.

Reply:
:blush: (thank you for proving yet again the value of code review)

Reply:
We should have tests around the RegistryClient.  Maybe can make `TestWebServer` take a list of expect-response pairs.